openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/NetCDFCollection.scala

```diff
@@ -28,99 +27,130 @@ object NetCDFCollection {
-    if(crs.length != 1) {
-      throw new IllegalArgumentException("All items in a collection must have the same CRS")
-    }
+  private def loadEmptyCollection(polygons: ProjectedPolygons, from_date: String, dataCubeParameters: DataCubeParameters, sc: SparkContext) = {
+    val boundingBox = polygons.extent
 
-    val bboxWGS84: Extent = items.map(_.bbox).reduce((a, b)=>(a.combine(b)))
+    val from = ZonedDateTime.parse(from_date, ISO_OFFSET_DATE_TIME)
+    val to = ZonedDateTime.parse("2025-06-01T00:00Z", ISO_OFFSET_DATE_TIME)
 
+    val scheme = FloatingLayoutScheme(dataCubeParameters.tileSize)
+    val maxSpatialResolution = CellSize(10, 10)
+    val multiple_polygons_flag = polygons.polygons.length > 1
 
-    val features: RDD[(TemporalProjectedExtent, MultibandTile)] = items.repartition(stacItems.length).flatMap(f=>{
-      val allTiles = f.links.flatMap(l=>{
-        l.bandNames.get.flatMap(b=> {
-          var gdalNetCDFLink = s"${l.href.toString.replace("file:", "NETCDF:")}:${b}"
-          if(!gdalNetCDFLink.startsWith("NETCDF:")) {
-            gdalNetCDFLink = s"NETCDF:${gdalNetCDFLink}"
-          }
-          try{
+    val metadata = DatacubeSupport.layerMetadata(
 
-            val rs = GDALRasterSource(gdalNetCDFLink,new GDALWarpOptions(outputFormat = None))
+      boundingBox, from, to, 0, ShortUserDefinedNoDataCellType(32767), scheme, maxSpatialResolution,
+      dataCubeParameters.globalExtent, multiple_polygons_flag
+    )
+    ContextRDD(sc.emptyRDD[(SpaceTimeKey, MultibandTile)], metadata)
+  }
 
-            /**
-             * Retrieving metadata using dataset directly, because sometimes metadata is so large that it doesn't fit the array allocated by GDALWarp
-             */
-            val units = rs.dataset.getMetadataItem("t#units",DefaultDomain,0)
-            val conventions: String = rs.dataset.getMetadataItem("NC_GLOBAL#Conventions",DefaultDomain,0)
-            val extraDim = rs.dataset.getMetadataItem("NETCDF_DIM_EXTRA",DefaultDomain,0)
+  private def loadCollection(stacItems: Seq[OpenSearchResponses.Feature], sc: SparkContext): RDD[(SpaceTimeKey, MultibandTile)] with Metadata[TileLayerMetadata[SpaceTimeKey]] = {
+    val items = sc.parallelize(stacItems)
 
-            if (!conventions.startsWith("CF-1")) {
-              throw new IllegalArgumentException(s"Only netCDF files with CF-1.x conventions are supported by this openEO backend, but found ${conventions}.")
-            }
-            if (extraDim != "{t}") {
-              throw new IllegalArgumentException("Only netCDF files with a time dimension named 't' are supported by this openEO backend.")
-            }
-            if( units != "days since 1990-01-01") {
-              throw new IllegalArgumentException("Only netCDF files with a time dimension in 'days since 1990-01-01' are supported by this openEO backend.")
+    if (stacItems.isEmpty) {
+      ContextRDD(sc.emptyRDD[(SpaceTimeKey, MultibandTile)], null)
```

Comment:
Would this be an ok RDD? If metadata is indeed optional, this might be an easier path compared to `loadEmptyCollection`

Reply:
null is not an option